src/litserve/loops/base.py

```diff
@@ -277,39 +284,55 @@ def pre_setup(self, lit_api: LitAPI, spec: Optional[LitSpec]):
 
         original = lit_api.unbatch.__code__ is LitAPI.unbatch.__code__
         if not lit_api.stream and any([
-            inspect.isgeneratorfunction(lit_api.predict),
-            inspect.isgeneratorfunction(lit_api.encode_response),
+            inspect.isgeneratorfunction(lit_api.predict) or inspect.isasyncgenfunction(lit_api.predict),
+            inspect.isgeneratorfunction(lit_api.encode_response) or inspect.isasyncgenfunction(lit_api.encode_response),
```

Comment:
nit, no need to `or` them here. the global `any` is basically a `or` already

Reply:
thanks! nice catch

Reply:
leaving it this way though for readability